File: src/test/java/org/nette/neon/lexer/NeonLexer2Test.java
package org.nette.neon.lexer;

import com.intellij.lexer.Lexer;
import com.intellij.testFramework.UsefulTestCase;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 *
 */
public class NeonLexer2Test extends UsefulTestCase {

	@Test
	public void test01() {
		Lexer l = createLexer();
		l.start("key: 'val'");

		assertEquals(NeonTokenTypes.NEON_LITERAL, l.getTokenType());
		assertEquals(0, l.getTokenStart());
		assertEquals(3, l.getTokenEnd());
		assertEquals("key", l.getTokenText());
		l.advance();

		assertEquals(NeonTokenTypes.NEON_COLON, l.getTokenType());
		assertEquals(3, l.getTokenStart());
		assertEquals(4, l.getTokenEnd());
		assertEquals(":", l.getTokenText());
		l.advance();

		assertEquals(NeonTokenTypes.NEON_WHITESPACE, l.getTokenType());
		assertEquals(4, l.getTokenStart());
		assertEquals(5, l.getTokenEnd());
		assertEquals(" ", l.getTokenText());
		l.advance();

		assertEquals(NeonTokenTypes.NEON_STRING, l.getTokenType());
		assertEquals(5, l.getTokenStart());
		assertEquals(10, l.getTokenEnd());
		assertEquals("'val'", l.getTokenText());
		l.advance();

		assertEquals(null, l.getTokenType());
	}

	@Test
	public void test02() {
		Lexer l = createLexer();
		l.start("key: 'val'", 4, 5);

		assertEquals(NeonTokenTypes.NEON_INDENT, l.getTokenType());
		assertEquals(4, l.getTokenStart());
		assertEquals(5, l.getTokenEnd());
		assertEquals(" ", l.getTokenText());
		l.advance();

		assertEquals(null, l.getTokenType());
	}

	private Lexer createLexer() {
		return new NeonLexer();
	}

}

File: src/test/kotlin/org/nette/neon/lexer/NeonLexer2Test.kt
package org.nette.neon.lexer

import com.intellij.lexer.Lexer
import com.intellij.testFramework.UsefulTestCase
import junit.framework.TestCase
import org.junit.Test

class NeonLexer2Test : UsefulTestCase() {
    private fun createLexer(): Lexer {
        return NeonLexer
    }

    @Test
    fun test01() {
        val l = createLexer()
        l.start("key: 'val'")

        assertEquals(NeonTokenTypes.NEON_LITERAL, l.tokenType)
        TestCase.assertEquals(0, l.tokenStart)
        TestCase.assertEquals(3, l.tokenEnd)
        TestCase.assertEquals("key", l.tokenText)
        l.advance()

        assertEquals(NeonTokenTypes.NEON_COLON, l.tokenType)
        TestCase.assertEquals(3, l.tokenStart)
        TestCase.assertEquals(4, l.tokenEnd)
        TestCase.assertEquals(":", l.tokenText)
        l.advance()

        assertEquals(NeonTokenTypes.NEON_WHITESPACE, l.tokenType)
        TestCase.assertEquals(4, l.tokenStart)
        TestCase.assertEquals(5, l.tokenEnd)
        TestCase.assertEquals(" ", l.tokenText)
        l.advance()

        assertEquals(NeonTokenTypes.NEON_STRING, l.tokenType)
        TestCase.assertEquals(5, l.tokenStart)
        TestCase.assertEquals(10, l.tokenEnd)
        TestCase.assertEquals("'val'", l.tokenText)
        l.advance()

        assertEquals(null, l.tokenType)
    }

    @Test
    fun test02() {
        val l = createLexer()
        l.start("key: 'val'", 4, 5)

        assertEquals(NeonTokenTypes.NEON_INDENT, l.tokenType)
        TestCase.assertEquals(4, l.tokenStart)
        TestCase.assertEquals(5, l.tokenEnd)
        TestCase.assertEquals(" ", l.tokenText)
        l.advance()

        assertEquals(null, l.tokenType)
    }
}
